calculator.go

```diff
@@ -1,13 +1,91 @@
 // Package calculator provides a library for simple calculations in Go.
 package calculator
 
-// Add takes two numbers and returns the result of adding them together.
-func Add(a, b float64) float64 {
-return a + b
+import (
+	"errors"
+	"math"
+)
+
+var (
+	// errNotEnoughNumbers is returned by functions that expects
+	// minimum of two arguments.
+	errNotEnoughNumbers = errors.New("minimum two numbers are required")
+
+	errDivisionByZero     = errors.New("division by zero")
+	errSqrtNumberNegative = errors.New("sqrt from negative number")
+)
+
+// Add takes 2 or more numbers and returns the result of adding them together.
+// If the number of arguments is less than two the function will return an error.
+func Add(a ...float64) (float64, error) {
```

Comment:
This is okay, but means that it's possible to call `Add` with either zero or one parameter, neither of which makes sense! Is there a way to get the compiler to help us enforce that `Add` needs _at least two_ parameters?